docs/architecture.md

```diff
@@ -2,32 +2,76 @@
+### Example
 
-The View component of Flux is represented by the Presenter+View couples. The Presenter is responsible for listening to the appropriate Store observables, and firing Actions to their respective handlers.
+To fully understand the concept, it's useful to trace one user action through its lifecycle of use in the app. Following is a simplified description of how the filter field (or search box) on the main item list screen works.
 
-### Routing
+When a user enters text into the search field, the textfield binding[2] on the `ItemListView` emits an event to an observer on the `ItemListPresenter`. The `ItemListPresenter` dispatches a `ItemListFilterAction`, which is a simple struct with one property - `filteringText: String`. The struct does a round trip through the `ItemListDisplayActionHandler`, `Dispatcher`, and `ItemListDisplayStore` before getting combined with the most recent list of items. This combined `Observable` stream with both the text and the items filters the list of items and maps the filtered list into individual cell configurations. The view, on receiving the updated / filtered list, re-renders the list of items to only show the ones that the user is searching for.
 
-The special case in this scenario is view routing. To handle the view-changing component of the architecture, there will be a special Store observed by a ContainerView. This “containing” view will never be displayed to the user; rather, it will perform the role of listening for navigation-specific Actions & performing the necessary top-level navigation stack swapping or navigation stack manipulation.
+There are a few other listeners for `ItemListFilterAction`s; for example, the `Observable` bound to the `isHidden` property of the Cancel button in the search bar maps the `ItemListFilterAction` with a simple `!isEmpty` check -- if the `ItemListFilterAction.filteringText` is empty, the cancel button is hidden, and if not, it's displayed.
 
-### Async
+### Current `ActionHandler` technical debt / area for improvement
 
-Asynchronous pieces of work are not handled in the definition of Flux architecture. In keeping with the typical solution for this pattern, async pieces of work in Lockbox iOS will be bundled into Action handlers, keeping Stores simple in-memory repositories to track persistent data as required by the View components.
+In the current LockFlux implementation, there is a discrepancy in the ways that asynchronous work is done. In some cases, async work is done behind the scenes at the `Store` level, and in others, as part of the pass-through setup between `ActionHandler`s and the `Dispatcher`. Ideally, we would get rid of the `ActionHandler` concept altogether, and `Presenter`s would construct and dispatch `Action`s directly to the `Dispatcher`. This will simplify tests and the architecture quite a bit.
 
 ---
 
 [1] the name here is pure semantics -- can be thought of as a ViewModel
```

Comment:
(realizing this is existing, but still ...)

one the preceding line, add a `<a name="note-1"/>` to match the end-note reference above (similar with the proceeding line).